documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -594,6 +611,106 @@ You can find the complete example in the
 https://github.com/mapstruct/mapstruct-examples/tree/master/mapstruct-field-mapping[mapstruct-examples-field-mapping]
 project on GitHub.
 
+[[mapping-with-builders]]
+=== Using builders
+
+MapStruct also supports mapping of immutable types via builders.
+When performing a mapping MapStruct checks if there is a builder for the type being mapped.
+This is done via the `BuilderProvider` SPI.
+If a Builder exists for a certain type, than that builder will be used for the mappings.
+
+The default implementation of the `BuilderProvider` requires the following for a type to have a builder:
+
+* The type (`Person`) has a parameterless public static method (builder creation method) that returns a type (`PersonBuilder`) different than the one that needs building
+* The found builder type (`PersonBuilder`) has a parameterless public method (build method) that returns the type being build (`Person`)
+
+If such type is found then MapStruct will use that type to perform the mapping to (i.e. it will look for setters into that type).
+If no factory method exists for instantiating the builder type the builder creation method would be used.
+To finish the mapping MapStruct will invoke the build method of the builder.
```

Comment:
To finish the mapping MapStruct generates code that will invoke the build method of the builder.